src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -374,6 +376,59 @@ public void sortFriendList(Comparator<Friend> comparator) {
         observableDirectory.setObservableDirectory(directory);
     }
 
+    /**
+     * Checks if accommodation object date is within the model's travel plan start and end date.
+     * @return false if accommodation start date and end date is within travel plan start date and end date.
+     */
+    public boolean isValidAccommodationDate(WanderlustDate startDate, WanderlustDate endDate) {
+        WanderlustDate travelPlanStartDate = directory.getStartDate();
+        WanderlustDate travelPlanEndDate = directory.getEndDate();
+
+        if (startDate.getValue().isBefore(travelPlanStartDate.getValue())
+                || startDate.getValue().isAfter(travelPlanEndDate.getValue())) {
+            return false;
+        }
+
+        if (endDate.getValue().isBefore(travelPlanStartDate.getValue())
+                || endDate.getValue().isAfter(travelPlanEndDate.getValue())) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Checks if activity object date time is within the model's travel plan start and end date.
+     * @return true if activity date is within travel plan start date and end date range.
+     */
+    public boolean isValidActivityDate(WanderlustDateTime activityDateTime) {
+        WanderlustDate travelPlanStartDate = directory.getStartDate();
+        WanderlustDate travelPlanEndDate = directory.getEndDate();
+
+        if (activityDateTime.getValue().toLocalDate().isBefore(travelPlanStartDate.getValue())
```

Comment:
Same for all date checking logic in this class

Reply:
fixed